### ISSUE TITLE
"ACL" for Users 

### ISSUE THREAD

**Issue by tomz1206** [nbuchwitz/icingaweb2-module-map#114]
Hello all,

is there a way to restrict the view of a user?
The user itself can only see some hosts in the host overview, but in the map, all hosts are visible.

If it´s not possible at the moment, it would be great if this feature will be implemented.

Many Thanks

**Comment by nbuchwitz** [comment]
The visible objects should be filtered by the user's permission. Please provide more details about your setup (see issue template).

**Comment by tomz1206** [comment]
The user has the following permissions under Roles -> icingadb:
icingadb/​filter/​objects: host.name=*Hostname1*|host.name=*Hostname2*
icingadb/​filter/​hosts: hostgroup.name=*Hostgroupname1*

So the user only sees the hosts and hostsgroups it should see, in the hosts and services overview.
Also the graphite module shows only these hosts and services.

If i append the following to the URL, i see what i should see: &hostgroup.name=*Hostgroupname1*
But unfortunately that´s not the solution we need.

* Icinga Web 2 Map version (System - About): 1.1.0
* Icinga Web 2 version and modules (System - About): 2.11.3
* Icinga DB Version: 1.0.2
* Version used (`icinga2 --version`): r2.13.6-1
* Operating System and version: Debian 11

Icinga Setup: Two Masters, seperated Web-Server, two redis Server, two icingadb-server and a galera-cluster.

I hope this information helps you, if not please let me know what´s missing. Many Thanks in advance.

**Comment by nbuchwitz** [comment]
Hm, the ACL filter worked with the traditional database. Looks like a regression with IcingaBD. @sukhwinder33445 do you have an idea?

**Comment by tomz1206** [comment]
Hello all,

Do we may have any news on this case?
Would be really great.

**Comment by nbuchwitz** [comment]
Haven't found the time to look into this. So no update yet.

**Comment by aledr** [comment]
My co-workers provided this temporary/possible fix to evaluate:

```
diff --git a/application/controllers/DataController.php b/application/controllers/DataController.php
index b184865..2088193 100644
--- a/application/controllers/DataController.php
+++ b/application/controllers/DataController.php
@@ -239,8 +239,10 @@ class DataController extends MapController
         $hostQuery->setResultSetClass(VolatileStateResults::class);
         $hostQuery->filter(IplFilter::like('host.vars.geolocation', '*'));
         //TODO not working properly
-        $this->Filter($hostQuery, $this->getFilter());
-        $hostQuery = $hostQuery->execute();
+       //$this->Filter($hostQuery, $this->getFilter());
+       $this->icingadbUtils->applyRestrictions($hostQuery);
+
+       $hostQuery = $hostQuery->execute();
         if ($hostQuery->hasResult()) {
             foreach ($hostQuery as $row) {
                 $hostname = $row->name;
@@ -272,7 +274,8 @@ class DataController extends MapController
             $serviceQuery->Filter(IplFilter::equal('state.is_problem', 'y'));
         }

-        $this->Filter($serviceQuery, $this->getFilter());
+       //$this->Filter($serviceQuery, $this->getFilter());
+       $this->icingadbUtils->applyRestrictions($serviceQuery);
         $serviceQuery = $serviceQuery->execute();
         if ($serviceQuery->hasResult()) {
             foreach ($serviceQuery as $row) {
@@ -306,7 +309,8 @@ class DataController extends MapController
             $serviceQuery->Filter(IplFilter::equal('service.state.is_problem', 'y'));
         }

-        $this->Filter($serviceQuery, $this->getFilter());
+       //$this->Filter($serviceQuery, $this->GetFilter());
+       $this->icingadbUtils->applyRestrictions($serviceQuery);
         $serviceQuery = $serviceQuery->execute();
         if ($serviceQuery->hasResult()) {
             foreach ($serviceQuery as $row) {
diff --git a/application/controllers/SearchController.php b/application/controllers/SearchController.php
index dbe5a8e..1660983 100644
--- a/application/controllers/SearchController.php
+++ b/application/controllers/SearchController.php
@@ -189,7 +189,8 @@ class SearchController extends MapController

         $query->limit($this->limit);
         //TODO not working properly
-        //$this->Filter($serviceQuery, $this->getFilter());
+       //$this->Filter($serviceQuery, $this->getFilter());
+       $this->icingadbUtils->applyRestrictions($serviceQuery);

         $query = $query->execute();

@@ -247,4 +248,4 @@ class SearchController extends MapController
         print $callback . "(" . json_encode($results) . ");";
         exit();
     }
-}
\ No newline at end of file
+}
diff --git a/configuration.php b/configuration.php
index 0ac8fe2..3075728 100644
--- a/configuration.php
+++ b/configuration.php
@@ -8,6 +8,15 @@ $mapModule = $section->add(N_($this->translate('Default map')), array(
     'priority' => 10
 ));

+// Custom Maps
+// Link State - General
+$mapModule = $section->add(N_($this->translate('Link State - General')), array(
+    'icon' => 'globe',
+    'description' => $this->translate('Visualize General Link State'),
+    'url' => 'map?servicegroup.name=link-state-general',
+    'priority' => 11
+));
+
 // stylesheets
 $this->provideCssFile('vendor/leaflet.css');
 $this->provideCssFile('vendor/MarkerCluster.css'); 
 ```

**Comment by nbuchwitz** [comment]
Please have a look if this is fixed by #116  (thanks @sukhwinder33445 )
